WordPress/src/main/java/org/wordpress/android/ui/activitylog/list/filter/ActivityLogTypeFilterViewModel.kt

```diff
@@ -41,46 +45,51 @@ class ActivityLogTypeFilterViewModel @Inject constructor(
     fun start(
         remoteSiteId: RemoteId,
         parentViewModel: ActivityLogViewModel,
-        initialSelection: List<Int>
+        dateRange: DateRange?,
+        initialSelection: List<String>
     ) {
         if (isStarted) return
         isStarted = true
         this.remoteSiteId = remoteSiteId
         this.parentViewModel = parentViewModel
         this.initialSelection = initialSelection
-
+        this.dateRange = dateRange
         fetchAvailableActivityTypes()
     }
 
     private fun fetchAvailableActivityTypes() {
         launch {
             _uiState.value = FullscreenLoading
-            val response = dummyActivityTypesProvider.fetchAvailableActivityTypes(remoteSiteId.value)
+            val response = activityLogStore.fetchActivityTypes(
+                    FetchActivityTypesPayload(
+                            remoteSiteId.value,
+                            dateRange?.first?.let { Date(it) },
+                            dateRange?.second?.let { Date(it) }
+                    )
+            )
             if (response.isError) {
                 _uiState.value = buildErrorUiState()
             } else {
-                _uiState.value = buildContentUiState(response.activityTypes)
+                _uiState.value = buildContentUiState(response.activityTypeModels)
             }
         }
     }
 
     private fun buildErrorUiState() =
             UiState.Error(Action(UiStringRes(R.string.retry)).apply { action = ::onRetryClicked })
 
-    private suspend fun buildContentUiState(activityTypes: List<DummyActivityType>): Content {
+    private suspend fun buildContentUiState(activityTypes: List<ActivityTypeModel>): Content {
         return withContext(bgDispatcher) {
-            // TODO malinjir replace the hardcoded header title
             val headerListItem = ListItemUiState.SectionHeader(
                     UiStringRes(R.string.activity_log_activity_type_filter_header)
             )
-            // TODO malinjir replace "it.toString()" with activity type name
             val activityTypeListItems: List<ListItemUiState.ActivityType> = activityTypes
                     .map {
                         ListItemUiState.ActivityType(
-                                id = it.id,
-                                title = UiStringText(it.toString()),
-                                onClick = { onItemClicked(it.id) },
-                                checked = initialSelection.contains(it.id)
+                                id = it.key,
+                                title = UiStringText("${it.name} (${it.count})"),
```

Comment:
Idea (💡 ): Consider removing the corresponding to the change TODO comment with the commit that actually removes the TODO. For example, the `Use real getActivityTypes endpoint` commit removed the TODO, but the actual change happened few commits later, this confused me as I was trying to figure out why this comment was removed for a couple of commits as I was going through review.

PS: By making every commit do exactly one think, this kind of problems are avoided and it helps the reviewer read the PR story as it was being written by the author, thus helping both sides. This way, the reviewer can review better and the author benefit from that ,more concrete, review.

Reply:
👍 makes sense, I'll try to follow this ;). Thanks!